yellowbrick/classifier/confusion_matrix.py

```diff
@@ -103,6 +108,20 @@ def score(self, X, y, sample_weight=None, percent=True):
         """
         y_pred = self.predict(X)
 
+
+        if self.label_mapping:
+            # if a mapping is passed to class apply it here.
+            y = character_mapping(y, self.label_mapping)
+            y_pred = character_mapping(y_pred, self.label_mapping)
+
+        if len(set(y_pred)) == len(set(self.classes_)):
+            if set(y_pred) != set(self.classes_):
+                self._label_encoder = LabelEncoder()
+                self._label_encoder.fit(self.classes_)
+
+                y = self._label_encoder.inverse_transform(y)
+                y_pred = self._label_encoder.inverse_transform(y_pred)
```

Comment:
@bbengfort I made several changes to the code from the first code review. I was wondering if you could take a look at the new code and see if this comment is still relevant.